include/sys/bus.h

```diff
@@ -145,17 +142,7 @@ struct bus_methods {
   void (*deactivate_resource)(device_t *dev, res_type_t type, resource_t *r);
 };
 
-#define BUS_METHODS(dev) (*(bus_methods_t *)(dev)->driver->interfaces[DIF_BUS])
-
-static inline void bus_intr_setup(device_t *dev, resource_t *irq,
-                                  ih_filter_t *filter, ih_service_t *service,
-                                  void *arg, const char *name) {
-  BUS_METHODS(dev->parent).intr_setup(dev, irq, filter, service, arg, name);
-}
-
-static inline void bus_intr_teardown(device_t *dev, resource_t *irq) {
-  BUS_METHODS(dev->parent).intr_teardown(dev, irq);
-}
+#define BUS_METHODS(dev) ((bus_methods_t *)(dev)->driver->interfaces[DIF_BUS])
```

Comment:
I agree this is a better approach, but still you should not tackle two issues in the same PR. Please split such changes in the future.